dnsapi/dns_selfhost.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env sh
+#
+#       Author: Marvin Edeler
+#       Report Bugs here: https://github.com/Marvo2011/acme.sh/issues/1
+#	Last Edit: 17.02.2022
+
+DNS_CHALLENGE_PREFIX_ESCAPED="_acme-challenge\."
+
+dns_selfhost_add() {
+  fulldomain=$1
+  txt=$2
+  _info "Calling acme-dns on selfhost"
+  _debug fulldomain "$fulldomain"
+  _debug txtvalue "$txt"
+
+  SELFHOSTDNS_UPDATE_URL="https://selfhost.de/cgi-bin/api.pl"
+
+  # Get values, but don't save until we successfully validated
+  SELFHOSTDNS_USERNAME="${SELFHOSTDNS_USERNAME:-$(_readaccountconf_mutable SELFHOSTDNS_USERNAME)}"
+  SELFHOSTDNS_PASSWORD="${SELFHOSTDNS_PASSWORD:-$(_readaccountconf_mutable SELFHOSTDNS_PASSWORD)}"
+  # These values are domain dependent, so read them from there
+  _getdeployconf SELFHOSTDNS_MAP
```

Comment:
don't use `_getdeployconf`,  it's for deploy hooks only.

Reply:
ok, but the usage of _readdomainconf / _savedomainconf is allowed for this?
SELFHOSTDNS_MAP are only valid for the specific domain, so saving in the accountconf isn't possible and would break the renewal process

Reply:
make sense.

Reply:
Just for clarification:
Should there be an equivalent for "_mutable" for the "_read- /savedomainconf" functions (like _read- /_saveaccountconf_mutable )?
I don't quite understand the differenz, except the "SAVED_" prefix.